### PR TITLE
Add `DictOfItems` type for configuration options

### PR DESCRIPTION
**Pull request by pawamoy**
As discussed in https://github.com/mkdocs/mkdocs/discussions/3240.
Absolutely feel free to push to the branch directly.
I copied the tests for `ListOfItems` and adapted them.

### CODE REVIEW

File: mkdocs/config/config_options.py [hunk 1]
```diff
@@ -199,6 +200,58 @@ def run_validation(self, value: object) -> list[T]:
...
 
 
+class DictOfItems(Generic[T], BaseConfigOption[Dict[str, T]]):
+    """
+    Validates a dict of items. Keys are always strings.
+
+    E.g. for `config_options.DictOfItems(config_options.Type(int))` a valid item is `{"a": 1, "b": 2}`.
+    """
+
+    required: Union[bool, None] = None  # Only for subclasses to set.
```
Comment (oprypin) [inline mkdocs/config/config_options.py]: Maybe the branch was out of date when you copied this part :)
```suggestion
    required: bool | None = None  # Only for subclasses to set.
```

File: mkdocs/config/config_options.py [hunk 2]
```diff
@@ -199,6 +200,58 @@ def run_validation(self, value: object) -> list[T]:
         return [fake_config[k] for k in fake_keys]
 
 
+class DictOfItems(Generic[T], BaseConfigOption[Dict[str, T]]):
+    """
+    Validates a dict of items. Keys are always strings.
```
Comment (oprypin) [inline mkdocs/config/config_options.py:205]: Hmm it doesn't actually validate that the keys are strings. Could you add that, and a test?

**Review by oprypin — CHANGES_REQUESTED** [review]

File: mkdocs/config/config_options.py (hunk 2 above)
Comment (pawamoy) [reply mkdocs/config/config_options.py:205]: Something like this? :thinking: 

```python
for key_name in fake_config:
    Type(str).run_validation(key_name)
```

Comment (oprypin) [reply mkdocs/config/config_options.py:205]: Definitely isinstance

Comment (pawamoy) [reply mkdocs/config/config_options.py:205]: Done :slightly_smiling_face:

File: mkdocs/config/config_options.py (hunk 1 above)
Comment (pawamoy) [reply mkdocs/config/config_options.py]: Yeah most probably, thanks for fixing it!

**Review by oprypin — APPROVED** [review]
Thanks!

**Comment by oprypin** [comment]
> Is there a hatch script to format the code?

```bash
hatch run style:format
```

**Comment by pawamoy** [comment]
Thanks, found it in pyproject but wasn't sure how to run :+1:

**Pull request merged**